common/commonincludes.hpp

```diff
@@ -46,7 +46,12 @@
 #include <net/if.h>
 #include <stdarg.h>
 #include <math.h>
+
+#if __has_include(<termios.h>)
+#include <termios.h>
+#elif __has_include(<sys/termios.h>)
 #include <sys/termios.h>
+#endif
 
```

Comment:
Since has_include isn't standard, please wrap with macro to detect GCC/G++ 

```
#ifdef __GNUC__
#if __has_include(<termios.h>)
#include <termios.h>
#elif __has_include(<sys/termios.h>)
#include <sys/termios.h>
#endif
#else
#include <termios.h>
#endif
```